rosidl_typesupport_opensplice_cpp/rosidl_typesupport_opensplice_cpp/rosidl_generator_dds_idl_extension.py

```diff
@@ -30,10 +30,7 @@ def get_include_directives(spec, subfolders):
     # prepend directives to include opensplice specific idl file
     for field in spec.fields:
         if field.type.pkg_name == 'builtin_msgs':
-            includes[:0] = [
-                '#ifndef OSPL_IDL_COMPILER',
-                '#include <dds_dcps.idl>',
-                '#endif']
+            includes.insert(0, '#include <dds_dcps.idl>')
```

Comment:
The usage of `--dds-types` is deprecated and made the DDS builtin types magically available before.

The `ifndef` was a workaround for `idlpp` not finding the `dds_dcps.idl` file. But when correctly adding the template path as an include directory that is not necessary anymore.